awx/main/isolated/manager.py

```diff
@@ -40,6 +42,51 @@ def __init__(self, cancelled_callback=None, check_callback=None):
         self.idle_timeout = max(60, 2 * settings.AWX_ISOLATED_CONNECTION_TIMEOUT)
         self.started_at = None
         self.captured_command_artifact = False
+        self.instance = None
+
+    def build_inventory(self, hosts):
+        if self.instance and self.instance.is_containerized:
+            inventory = { 'all': { 'hosts': {} } }
+            for host in hosts:
+                inventory['all']['hosts'][host] = self.kubectl_or_oc_vars
+        else:
+            inventory = '\n'.join([
+                '{} ansible_ssh_user={}'.format(host, settings.AWX_ISOLATED_USERNAME)
+                for host in hosts
+            ])
+
+        return inventory
+
+    @property
+    def kubectl_or_oc_vars(self):
+        extravars = {}
+
+        credential = self.instance.instance_group.credential
+
+        extravars['ansible_connection'] = 'kubectl'
+        extravars['ansible_kubectl_host'] = credential.get_input('host')
+        extravars['ansible_kubectl_namespace'] = 'awx'
+
+        if credential.kind == 'kubernetes_bearer_token':
+            extravars['ansible_kubectl_token'] = credential.get_input('bearer_token')
```

Comment:
I'm not super familiar with our newer credential code. @ryanpetrello Does it matter that `bearer_token` is a secret variable? Specifically, do we have to explicitly decode it .. or does `get_input()` do that for us?

Reply:
`.get_input()` does it for us:

https://github.com/ansible/awx/blob/devel/awx/main/models/credential/__init__.py#L277